sympy/parsing/autolev/_listener_autolev_antlr.py

```diff
@@ -399,9 +399,58 @@ def writeImaginary(self, ctx):
     self.write(b)
     self.var_list = []
 
+
+def gcd(a, b):
+    if (b == 0):
+        return a
+    return gcd(b, a%b)
+
+
+# Returns smallest integer k such that k * str becomes
+# natural. str is an input floating point number
+def findnum(str):
+    import math as m
+    # reference: https://www.geeksforgeeks.org/smallest-number-multiply-convert-floating-point-natural/
+    # Find size of string representing a
+    # floating point number.
+    n = len(str)
+    # Below is used to find denominator in
+    # fraction form.
+    count_after_dot = 0
+
+    # Used to find value of count_after_dot
+    dot_seen = 0
+
+    # To find numerator in fraction form of
+    # given number. For example, for 30.25,
+    # numerator would be 3025.
+    num = 0
+    for i in range(n):
+        if (str[i] != '.'):
+            num = num*10 + int(str[i])
+            if (dot_seen == 1):
+                count_after_dot += 1
+        else:
+            dot_seen = 1
+
+    # If there was no dot, then number
+    # is already a natural.
+    if (dot_seen == 0):
+        return 1
+
+    # Find denominator in fraction form. For example,
+    # for 30.25, denominator is 100
+    dem = int(m.pow(10, count_after_dot))
+
+    # Result is denominator divided by
+    # GCD-of-numerator-and-denominator. For example, for
+    # 30.25, result is 100 / GCD(3025,100) = 100/25 = 4
+    return (dem / gcd(num, dem))
```

Comment:
Not sure what all the above code is for. Is it related to this PR?

Reply:
This is related to replacing np.linespace with range. This is because we can't use the range function with decimal input to create an auto-incrementing scale.

Reply:
So this is the python.math version of np.linspace?

Reply:
Yes, Converting decimals to integer by multiplying by a factor. This factor is used to create timescales with losing precision.